lineman/app/components/thread_page/thread_page_controller.coffee

```diff
@@ -1,24 +1,48 @@
 angular.module('loomioApp').controller 'ThreadPageController', ($scope, $routeParams, $location, $rootScope, Records, MessageChannelService, CurrentUser, DiscussionFormService, ScrollService) ->
   $rootScope.$broadcast('currentComponent', { page: 'threadPage'})
 
-  $scope.$on 'threadPageEventsLoaded',    (event, sequenceId) =>
-    ScrollService.scrollTo("#sequence-#{sequenceId}") if @focusMode == 'activity'
-  $scope.$on 'threadPageProposalsLoaded', (event, proposalId) =>
-    ScrollService.scrollTo("#proposal-#{proposalId}") if @focusMode == 'proposal'
-
-  Records.discussions.findOrFetchByKey($routeParams.key).then (discussion) =>
-    @discussion = discussion
-    $rootScope.$broadcast('setTitle', @discussion.title)
-    @group = @discussion.group()
-    $rootScope.$broadcast('viewingThread', @discussion)
-    MessageChannelService.subscribeTo("/discussion-#{@discussion.key}", @onMessageReceived)
-    @setFocusMode()
-    ScrollService.scrollTo('.thread-context') if @focusMode == 'context'
-  , (error) ->
+  @performScroll = ->
+    elementToFocus = @elementToFocus()
+    if elementToFocus && !@scrolledAlready
+      ScrollService.scrollTo elementToFocus
+      @scrolledAlready = true
+
+  @elementToFocus = ->
+    if $location.hash().match(/^proposal-\d+$/) and Records.proposals.find(@focusedProposalId)
+      "#proposal-#{@focusedProposalId}"
+    else if @discussion.lastSequenceId == 0 or @sequenceIdToFocus == -1
+      ".thread-context"
+    else if Records.events.findByDiscussionAndSequenceId(@discussion, @sequenceIdToFocus)
+      if @discussion.isUnread()
+        ".activity-card__new-activity"
+      else
+        "#sequence-#{@sequenceIdToFocus}"
+    else
+      false # our record isn't there yet
+
+  @init = (discussion) =>
+    if discussion and !@discussion?
+      @discussion = discussion
+      @group = @discussion.group()
+      @sequenceIdToFocus = @discussion.reader().lastReadSequenceId # or location hash when we put it back in.
```

Comment:
I think it might be nice to add a last item element.. that does the same kinda thing as new activity but is not visible.